subsys/bluetooth/host/l2cap.c

```diff
@@ -2991,6 +3002,30 @@ int bt_l2cap_ecred_chan_reconfigure(struct bt_l2cap_chan **chans, uint16_t mtu)
 		return -EBUSY;
 	}
 
+	/* default (max) MPS for channels */
+	if (mps == 0) {
+		mps = BT_L2CAP_RX_MTU;
+	}
+
+	/* MPS shall not be bigger than MTU + BT_L2CAP_SDU_HDR_SIZE
+	 * as the remaining bytes cannot be used.
+	 */
+	mps = MIN(mtu + BT_L2CAP_SDU_HDR_SIZE, mps);
+
+	for (int j = 0; j < i; j++) {
+		ch = BT_L2CAP_LE_CHAN(chans[j]);
+		/* For now simply don't allow MPS changes on channels using
+		 * segment direct API
+		 *
+		 * TODO we could be smarted here
```

Comment:
```suggestion
		 * TODO we could be smarter here
```

By doing what? A TODO should be a bit more elaborate IMO :)

Reply:
be smarter to allow for MPS change on those channels too :)
in some cases we could allow to reduce MPS (and with some additional tracking increase it)